src/data/list/basic.lean

```diff
@@ -1363,6 +1368,10 @@ begin
     cases x, simpa using hxy, simp at hxy, simp [y_ih hxy.2, h hxy.1] }
 end
 
+theorem map_filter_eq_foldr (f : α → β) (p : α → Prop) [decidable_pred p] (as : list α)
+  : map f (filter p as) = foldr (λ a bs, if p a then f a :: bs else bs) [] as :=
+by induction as; [ refl, simp! [*, apply_ite (map f)] ]
```

Comment:
Please don't use `X; [ Y, Z ]` when you could just write `{ X, { Y }, { Z } }`. It's unnecessarily obfuscatory.

Reply:
Fixed throughout.